glib-macros/src/properties.rs

```diff
@@ -0,0 +1,631 @@
+        let span = p.attrs_span;
+        quote_spanned!(span=> pub fn #fn_ident<F: Fn(&Self) + 'static>(&self, f: F) -> #crate_ident::SignalHandlerId {
+            self.connect_notify_local(Some(#name), move |this, _| {
+                f(this)
+            })
+        })
+    });
+    quote!(#(#connection_fns)*)
+}
+
+fn expand_emit_impl(props: &[PropDesc]) -> TokenStream2 {
+    let crate_ident = crate_ident_new();
+    let emit_fns = props.iter().map(|p| {
+        let name = &p.name;
+        let fn_ident = format_ident!("emit_{}", name_to_ident(name));
+        let span = p.attrs_span;
+        let enum_ident = name_to_enum_ident(name.value());
+        quote_spanned!(span=> pub fn #fn_ident(&self) {
+            self.notify_by_pspec(
+                &<<Self as #crate_ident::object::ObjectSubclassIs>::Subclass
+                    as #crate_ident::subclass::object::DerivedObjectProperties>::derived_properties()
+                [DerivedPropertiesEnum::#enum_ident as usize]
+            );
+        })
+    });
+    quote!(#(#emit_fns)*)
+}
+
+fn name_to_enum_ident(mut name: String) -> syn::Ident {
+    let mut slice = name.as_mut_str();
+    while let Some(i) = slice.find('-') {
+        let (head, tail) = slice.split_at_mut(i);
+        if let Some(c) = head.get_mut(0..1) {
+            c.make_ascii_uppercase();
+        }
+        slice = &mut tail[1..];
+    }
+    if let Some(c) = slice.get_mut(0..1) {
+        c.make_ascii_uppercase();
+    }
+    let enum_member: String = name.split('-').collect();
+    format_ident!("{}", enum_member)
+}
+
+fn expand_properties_enum(props: &[PropDesc]) -> TokenStream2 {
+    let properties: Vec<syn::Ident> = props
+        .iter()
+        .map(|p| {
+            let name: String = p.name.value();
+            name_to_enum_ident(name)
+        })
+        .collect();
+    let props = properties.iter();
+    let indices = 0..properties.len();
+    quote! {
+        #[repr(usize)]
+        #[derive(Debug, Copy, Clone)]
+        enum DerivedPropertiesEnum {
+            #(#props,)*
+        }
+        impl std::convert::TryFrom<usize> for DerivedPropertiesEnum {
+            type Error = usize;
+
+            fn try_from(item: usize) -> Result<Self, Self::Error> {
+                match item {
+                    #(#indices => Ok(Self::#properties),)*
+                    _ => Err(item)
+                }
+            }
+        }
+    }
+}
+
+pub fn impl_derive_props(input: PropsMacroInput) -> TokenStream {
+    let struct_ident = &input.ident;
+    let crate_ident = crate_ident_new();
+    let wrapper_type = input.wrapper_ty;
+    let fn_properties = expand_properties_fn(&input.props);
+    let fn_property = expand_property_fn(&input.props);
+    let fn_set_property = expand_set_property_fn(&input.props);
+    let getset_properties_impl = expand_getset_properties_impl(&input.props);
+    let connect_prop_notify_impl = expand_connect_prop_notify_impl(&input.props);
```

Comment:
This should also generate `notify_$property` function calls that do `ObjectExt::notify(self, $property)`

Reply:
`notify_by_pspec()` please :)

Reply:
@bilelmoussaoui this one is pending

Reply:
It was already there but named emit_impls wrongly,